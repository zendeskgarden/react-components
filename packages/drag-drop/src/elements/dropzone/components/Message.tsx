/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, HTMLAttributes, useEffect } from 'react';
import { StyledMessage } from '../../../styled';
import { useDropzoneContext } from '../../../utils/useDropzoneContext';

/**
 * @extends HTMLAttributes<HTMLParagraphElement>
 */
export const Message = forwardRef<HTMLParagraphElement, HTMLAttributes<HTMLParagraphElement>>(
  (props, ref) => {
    const { setHasMessage, hasMessage } = useDropzoneContext();

    useEffect(() => {
      if (setHasMessage && !hasMessage) {
        setHasMessage(true);
      }

      return () => {
        if (setHasMessage && hasMessage) {
          setHasMessage(false);
        }
      };
    }, [setHasMessage, hasMessage]);

    return <StyledMessage {...props} ref={ref} />;
  }
);

Message.displayName = 'Dropzone.Message';
