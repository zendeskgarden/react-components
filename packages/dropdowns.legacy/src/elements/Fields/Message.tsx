/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React from 'react';
import PropTypes from 'prop-types';
import { IMessageProps, Message as FormMessage, VALIDATION } from '@zendeskgarden/react-forms';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Message = React.forwardRef<HTMLDivElement, IMessageProps>((props, ref) => (
  <FormMessage ref={ref} {...props} />
));

Message.displayName = 'Message';

Message.propTypes = {
  validation: PropTypes.oneOf(VALIDATION)
};
