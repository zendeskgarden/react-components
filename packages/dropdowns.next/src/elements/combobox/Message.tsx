/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import { VALIDATION } from '@zendeskgarden/react-forms';
import { IMessageProps } from '../../types';
import useFieldContext from '../../context/useFieldContext';
import { StyledMessage } from '../../views';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Message = forwardRef<HTMLDivElement, IMessageProps>((props, ref) => {
  const { hasMessage, setHasMessage } = useFieldContext();

  useEffect(() => {
    // context callback
    setHasMessage(true);

    return () => setHasMessage(false);
  }, [hasMessage, setHasMessage]);

  return <StyledMessage {...props} ref={ref} />;
});

Message.displayName = 'Message';

Message.propTypes = {
  validation: PropTypes.oneOf(VALIDATION),
  validationLabel: PropTypes.string
};
