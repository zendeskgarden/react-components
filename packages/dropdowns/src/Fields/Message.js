/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import { StyledMessage } from '../styled';

const VALIDATION = {
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
  NONE: 'none'
};

/**
 * Accepts all `<div>` props. Must be nested with a `<Field>` component.
 */
const Message = StyledMessage;

Message.displayName = 'Message';

Message.propTypes = {
  validation: PropTypes.oneOf([
    VALIDATION.SUCCESS,
    VALIDATION.WARNING,
    VALIDATION.ERROR,
    VALIDATION.NONE
  ])
};

export default Message;
