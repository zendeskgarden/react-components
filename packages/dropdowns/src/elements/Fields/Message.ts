/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import { StyledMessage, VALIDATION } from '../../styled';
import { HTMLAttributes } from 'react';

export interface IMessageProps extends HTMLAttributes<HTMLDivElement> {
  validation?: VALIDATION;
}

/**
 * Accepts all `<div>` props. Must be nested with a `<Field>` component.
 */
const Message = StyledMessage as React.FunctionComponent<IMessageProps>;

Message.propTypes = {
  validation: PropTypes.oneOf([VALIDATION.SUCCESS, VALIDATION.WARNING, VALIDATION.ERROR])
};

export default Message;
