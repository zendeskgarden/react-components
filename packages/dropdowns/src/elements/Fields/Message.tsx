/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { HTMLAttributes } from 'react';
import PropTypes from 'prop-types';
import { Message as FormMessage } from '@zendeskgarden/react-forms';
import { VALIDATION } from '../../utils/validation';

export interface IMessageProps extends HTMLAttributes<HTMLDivElement> {
  validation?: VALIDATION;
}

/**
 * Accepts all `<div>` props. Must be nested with a `<Field>` component.
 */
export const Message = React.forwardRef<HTMLDivElement, IMessageProps>((props, ref) => (
  <FormMessage ref={ref} {...props} />
));

Message.displayName = 'Message';

Message.propTypes = {
  validation: PropTypes.oneOf(['success', 'warning', 'error'])
};
