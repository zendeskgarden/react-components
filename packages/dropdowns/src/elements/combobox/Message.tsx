/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import React, { forwardRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import { VALIDATION } from '@zendeskgarden/react-forms';
import { IMessageProps } from '../../types';
import useFieldContext from '../../context/useFieldContext';
import { StyledMessage } from '../../views';

/**
 * @extends HTMLAttributes<HTMLDivElement>
 */
export const Message = forwardRef<HTMLDivElement, IMessageProps>((props, ref) => {
  const { messageProps, setHasMessage } = useFieldContext();

  useEffect(() => {
    setHasMessage(true);

    return () => setHasMessage(false);
  }, [setHasMessage]);

  return <StyledMessage {...messageProps} {...props} ref={ref} />;
});

Message.displayName = 'Message';

Message.propTypes = {
  validation: PropTypes.oneOf(VALIDATION),
  validationLabel: PropTypes.string
};
