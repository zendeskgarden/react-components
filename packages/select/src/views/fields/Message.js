/**
 * Copyright Zendesk, Inc.
 *
 * Use of this source code is governed under the Apache License, Version 2.0
 * found at http://www.apache.org/licenses/LICENSE-2.0.
 */

import PropTypes from 'prop-types';
import { retrieveTheme } from '@zendeskgarden/react-theming';
import { Message as TextfieldMessage } from '@zendeskgarden/react-textfields';

const COMPONENT_ID = 'select.message';
const VALIDATION = {
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error'
};

/**
 * Accepts all `<div>` props
 */
const Message = TextfieldMessage.extend.attrs({
  'data-garden-id': COMPONENT_ID,
  'data-garden-version': PACKAGE_VERSION
})`
  ${props => retrieveTheme(COMPONENT_ID, props)};
`;

Message.propTypes = {
  validation: PropTypes.oneOf([VALIDATION.SUCCESS, VALIDATION.WARNING, VALIDATION.ERROR])
};

/** @component */
export default Message;
