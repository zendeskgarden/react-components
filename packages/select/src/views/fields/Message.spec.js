import React from 'react';
import { shallow } from 'enzyme';
import Message from './Message';

describe('Message', () => {
  it('renders default styling', () => {
    const wrapper = shallow(<Message />);

    expect(wrapper).toMatchSnapshot();
  });
});
