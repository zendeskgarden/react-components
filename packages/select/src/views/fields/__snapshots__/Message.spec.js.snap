// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message renders default styling 1`] = `
<div
  className="c-txt__message "
  data-garden-id="select.message"
  data-garden-version="version"
/>
`;
